### PR TITLE
update titiler

### PR DESCRIPTION
**Pull request by vincentsarago**
This pr does:
- Update titiler and titiler-pgstac version
- Update rasterio >=1.3 (and remove `pctiler/pctiler/reader_cog.py`)

#### Frontend changes

- remove `asset_expression` (Mosaic and Item)
- histogram band names are prefixed with `b` (e.g `b1`) (Mosaic and Item) (ref: https://github.com/cogeotiff/rio-tiler/blob/main/docs/src/v4_migration.md#band-names)
- expression for STAC have to be in form of `{asset}_b{band_name}` (e.g `red_b1/green_b1`) (Mosaic and Item) (ref: https://github.com/cogeotiff/rio-tiler/blob/main/docs/src/v4_migration.md#multibasereader-expressions)
- added `asset_as_band` option to force expression to be in form of `{asset}` (e.g `red/green`) (Mosaic and Item)
- expression's band should now be delimited with `;` (previously `,` was accepted) (Mosaic and Item)
- point output model to include band_names (Item)
- added `algorithm` options

Ref: https://github.com/stac-utils/titiler-pgstac/blob/master/CHANGES.md?plain=1#L55-L63

### CODE REVIEW

File: pctiler/pctiler/endpoints/item.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
     colormap_dependency=PCColorMapParams,
     reader_dependency=ReaderParams,
     router_prefix=get_settings().item_endpoint_prefix,
+    # We remove the titiler default `/map` viewer
+    add_viewer=False,
```
Comment (vincentsarago) [inline pctiler/pctiler/endpoints/item.py:34]: in titiler 0.8 we added a `/map` viewer which is added by default with the Tiler Factories

File: pctiler/pctiler/endpoints/pg_mosaic.py [hunk 2]
```diff
@@ -39,13 +39,16 @@ def __init__(self, request: Request):
     reader_dependency=ReaderParams,
     router_prefix=get_settings().mosaic_endpoint_prefix,
     backend_dependency=BackendParams,
+    add_map_viewer=False,
+    add_statistics=False,
+    add_mosaic_list=False,
```
Comment (vincentsarago) [inline pctiler/pctiler/endpoints/pg_mosaic.py:44]: those are already set to False in the TileFactory so it's mostly informative
https://github.com/stac-utils/titiler-pgstac/blob/master/titiler/pgstac/factory.py#L111-L116

File: pctiler/pctiler/endpoints/pg_mosaic.py [hunk 3]
```diff
@@ -39,13 +39,16 @@ def __init__(self, request: Request):
...
 )
 
 
 @pgstac_mosaic_factory.router.get(
     "/info", response_model=MosaicInfo, response_class=ORJSONResponse
 )
-def map(
+def mosaic_info(
```
Comment (vincentsarago) [inline pctiler/pctiler/endpoints/pg_mosaic.py:51]: renamed for consistency

File: pctiler/pctiler/reader.py [hunk 4]
```diff
@@ -44,36 +44,41 @@ def __init__(self, request: Request):
...
 
-        return asset_url
+        asset_info = self.input.assets[asset]
+        info = AssetInfo(url=asset_url)
+
+        if "file:header_size" in asset_info.extra_fields:
+            h = asset_info.extra_fields["file:header_size"]
+            info["env"] = {"GDAL_INGESTED_BYTES_AT_OPEN": h}
+
+        return info
```
Comment (vincentsarago) [inline pctiler/pctiler/reader.py:69]: we now return a TypedDict with the URL and a dictionary used to set some GDAL env (if available)

File: pccommon/setup.py [hunk 5]
```diff
@@ -6,9 +6,8 @@
 inst_reqs = [
     # --->
     # TODO: restore fastapi release install after starlette dep upgraded to >= 0.21.0
-    # "fastapi>=0.75.2",
-    "fastapi @ git+https://github.com/mmcfarland/fastapi/@982e7caf086bffeace8554da6d69e5f3082f14a3#egg=fastapi",
-    "starlette>=0.21.0,<0.22.0",
+    "fastapi>=0.87",
+    "starlette>=0.22.0,<0.23.0",
```
Comment (vincentsarago) [inline pccommon/setup.py:10]: @mmcfarland is it ok to use starlette 0.22

**Comment by mmcfarland** [comment]
Due to the breaking expression changes, we're going to wait to evaluate this after the next release (end of Jan).

**Comment by mmcfarland** [comment]
Thanks @vincentsarago , I'm ready to merge this. Could you resolve the conflict (I can't push to the fork)? The same dependency upgrades have merged after this PR was created.

**Comment by vincentsarago** [comment]
conflict resolved, I'll let you go ahead an merge

**Pull request merged**